### PR TITLE
add natvis file for visualstudio debugger.

### LINKED ISSUE #2530: add natvis file for visualstuduo user
### Expected Behavior:

STRING and GenericVector can debug user friendly.

ref: https://github.com/g-truc/glm/blob/master/util/glm.natvis
ref: https://docs.microsoft.com/en-us/visualstudio/debugger/create-custom-views-of-native-objects?view=vs-2019

### PR DESCRIPTION
**Pull request by zhuangzhuang**
fix #2530

### CODE REVIEW

File: util/tesseract.natvis [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <Expand>
+      <Item Name="[size]" ExcludeView="simple">size_used_</Item>
+      <Item Name="[capacity]" ExcludeView="simple">size_reserved_</Item>
+      <ArrayItems>
+        <Size>size_used_</Size>
+        <ValuePointer>data_</ValuePointer>
+      </ArrayItems>
+    </Expand>
+  </Type>
+</AutoVisualizer>
```
Comment (stweil) [inline util/tesseract.natvis]: Please add a terminating line feed.

**Comment by stweil** [comment]
You added a new directory `util` for the new file, but as far as I have understood Visual Studio allows any directory and filename. I'd prefer using an existing directory (maybe `doc/tesseract.natvis`) or having an invisible `.tesseract.natvis` in the root directory.

**Comment by stweil** [comment]
The longterm solution would be replacing proprietary data types by standard data types: `std::vector` instead of `GenericVector` and `std::string` instead of `STRING`.

**Comment by stweil** [comment]
Could you please also add links to the .natvis documentation either in the commit message or as a comment in the .natvis file (or to both)? That helps maintaining that file.

**Comment by egorpugin** [comment]
> The longterm solution would be replacing proprietary data types by standard data types: std::vector instead of GenericVector and std::string instead of STRING.

Can we do this for 5.0.0?
cc: @zdenop

File: util/tesseract.natvis (hunk 1 above)
Comment (zhuangzhuang) [reply util/tesseract.natvis]: has updated

**Comment by egorpugin** [comment]
Ideal place for natvis is `src/tesseract.natvis`.

**Comment by zhuangzhuang** [comment]
> Ideal place for natvis is `src/tesseract.natvis`.
There is no standard, refer to other projects.
imgui place in misc folder  https://github.com/ocornut/imgui/tree/master/misc/natvis
gsl place in root folder   https://github.com/microsoft/GSL
eigen place in debug folder  https://bitbucket.org/eigen/eigen/src/default/debug/msvc/

**Comment by stweil** [comment]
Thank you. I merged it now. We can move the file later as soon as we find a better place for it.

**Comment by stweil** [comment]
@zhuangzhuang, please [set your real name](https://help.github.com/en/articles/setting-your-username-in-git) in your Git configuration for future pull requests.

**Pull request merged**